File: dart/armstrong-numbers/lib/armstrong_numbers.dart
import 'dart:math';

class ArmstrongNumbers {
  bool isArmstrongNumber(int number) {
    final digits = _toDigits(number);
    final sum = digits.fold<int>(
        0, (prev, element) => prev + pow(element, digits.length) as int);
    return number == sum;
  }

  List<int> _toDigits(int number) {
    List<int> digits = [];
    int n = number;
    while (n != 0) {
      var digit = n % 10;
      digits.add(digit);
      n = n ~/ 10;
    }
    return digits;
  }
}

File: python/armstrong-numbers/armstrong_numbers.py
def _iterate_digits(number: int):
    assert number >= 0

    if number == 0:
        yield 0
    else:
        while number != 0:
            yield number % 10
            number = int(number / 10)


def is_armstrong_number(number: int) -> bool:
    digits = list(_iterate_digits(number))
    cnt_digits = len(digits)
    return number == sum(map(lambda d: d ** cnt_digits, digits))
